### PR TITLE
Add initial abstraction for 17cy

### PR DESCRIPTION
**Pull request by jlmaners**

### CODE REVIEW

**Comment by jlmaners** [comment]
@vanstinator here's an initial pass at things for 17CY generation based on the abstraction. The one thing that confused me a bit was whether or not an additional function needed to be implemented in vehicle.py to handle sending the commands? That's why I've left this in draft right now.

**Comment by vanstinator** [comment]
Awesome! I'm gonna take a good look at this today!

**Comment by vanstinator** [comment]
This looks great! 

Looking at the send command function I think we just need to update the function signature on the base abstract function to add the `value` bit that your implementation will need. And the `plus` generation will just ignore that parameter.

**Comment by jlmaners** [comment]
@vanstinator think this is finally ready to be reviewed and then I can work to validate it works with the HA component that's in alpha

File: toyota_na/vehicle/vehicle.py [hunk 1]
```diff
@@ -22,7 +23,20 @@ async def get_vehicles(client: ToyotaOneClient) -> list[ToyotaVehicle]:
...
+        ):
+            vehicle = SeventeenCYToyotaVehicle(
+                client=client,
+                has_remote_subscription=vehicle["remoteSubscriptionStatus"] == "ACTIVE",
+                model_name=vehicle["modelName"],
+                model_year=vehicle["modelYear"],
+                vin=vehicle["vin"],
+            )
+            await vehicle.update()
             vehicles.append(vehicle)
```
Comment (vanstinator) [inline toyota_na/vehicle/vehicle.py:40]: Let's pull these out of the if/else so we don't repeat these lines twice

**Review by vanstinator — APPROVED** [review]
This looks great. There's more duplicated code than I'd intended initially but like we chatted about in DM it's probably better to start this way. We can always merge more of the common stuff later :)

**Pull request merged**